clients/src/main/java/org/apache/kafka/clients/producer/internals/RecordAccumulator.java

```diff
@@ -263,7 +263,7 @@ public void reenqueue(RecordBatch batch, long now) {
      * <li>The accumulator has been closed
      * </ol>
      */
-    public ReadyCheckResult ready(Cluster cluster, long nowMs) {
+    public ReadyCheckResult ready(Cluster cluster, Set<TopicPartition> partitionsInFlight, long nowMs) {
```

Comment:
Does it make sense to make a version of this that matches the old contract so all the existing calls don't need to change? Something like:
```java
public ReadyCheckResult ready(Cluster cluster, long nowMs) {
   return ready(cluster, null, nowMs);
}
```